### PR TITLE
Fix Linter issues

### PR DESCRIPTION
**Pull request by mrlukasbos**
This PR fixes the linter issues that could not be fixed with an autofixer. 

I found some _really_ tricky patterns at some places. Lot's of shadowed variables, inline function definitions using a lot of local variables, and quite a number of 'for ... in' loops.

### CODE REVIEW

File: lib/HomeyCompose.js [hunk 1]
```diff
@@ -473,10 +469,14 @@ class HomeyCompose {
...
+        appLocales[appComposeLocaleLanguage] = deepmerge(appLocales[appComposeLocaleLanguage],
+          appComposeLocale);
       } else {
-        const value = objectPath.get(appLocales[appComposeLocaleLanguage], appComposeLocaleIdArray) || {};
-        objectPath.set(appLocales[appComposeLocaleLanguage], appComposeLocaleIdArray, deepmerge(value, appComposeLocale));
+        const value = objectPath.get(appLocales[appComposeLocaleLanguage], appComposeLocaleIdArray)
+          || {};
+        objectPath.set(appLocales[appComposeLocaleLanguage],
+          appComposeLocaleIdArray,
+          deepmerge(value, appComposeLocale));
```
Comment (tjallingt) [inline lib/HomeyCompose.js:479]:
```suggestion
        const value = objectPath.get(appLocales[appComposeLocaleLanguage], appComposeLocaleIdArray);

        objectPath.set(
          appLocales[appComposeLocaleLanguage],
          appComposeLocaleIdArray,
          deepmerge(value || {}, appComposeLocale),
        );
```

File: lib/HomeyCompose.js [hunk 2]
```diff
@@ -473,10 +469,14 @@ class HomeyCompose {
       appLocales[appComposeLocaleLanguage] = appLocales[appComposeLocaleLanguage] || {};
 
       if (appComposeLocaleIdArray.length === 0) {
-        appLocales[appComposeLocaleLanguage] = deepmerge(appLocales[appComposeLocaleLanguage], appComposeLocale);
+        appLocales[appComposeLocaleLanguage] = deepmerge(appLocales[appComposeLocaleLanguage],
+          appComposeLocale);
```
Comment (tjallingt) [inline lib/HomeyCompose.js:473]:
```suggestion
        appLocales[appComposeLocaleLanguage] =
          deepmerge(appLocales[appComposeLocaleLanguage], appComposeLocale);
```

Comment (tjallingt) [reply lib/HomeyCompose.js:473]: hope eslint accepts this

File: lib/HomeyCompose.js (hunk 1 above)
Comment (tjallingt) [reply lib/HomeyCompose.js:479]: hope eslint accepts this

File: lib/HomeyCompose.js (hunk 2 above)
Comment (mrlukasbos) [reply lib/HomeyCompose.js:473]: Nope EsLint did not like that

File: lib/HomeyCompose.js [hunk 3]
```diff
@@ -83,18 +85,49 @@ class HomeyCompose {
...
+      for (const childSettingId of Object.keys(settingObj.children)) {
+        this.extendSetting(settingsTemplates, settingObj.children[childSettingId]);
+      }
+    } else if (settingObj.$extends) {
+      const templateIds = [].concat(settingObj.$extends);
+
+      let settingTemplate = {};
+      let templateId;
+      for (const i of Object.keys(templateIds)) {
+        templateId = templateIds[i];
```
Comment (tjallingt) [inline lib/HomeyCompose.js:99]:
```suggestion
      for (const templateId of Object.values(templateIds)) {
```

File: lib/AthomApi.js [hunk 4]
```diff
@@ -41,7 +41,7 @@ class AthomApi {
   }
 
   async _initApi() {
-    if (this._api) return this._api;
+    if (this._api) ;
```
Comment (tjallingt) [inline lib/AthomApi.js:44]:
```suggestion
    if (this._api) return this._api;
```

File: lib/AthomApi.js [hunk 5]
```diff
@@ -81,10 +82,10 @@ class AthomApi {
...
           app.get('/auth', (req, res) => {
             res.sendFile(path.join(__dirname, '..', 'assets', '1px.png'));
-            const { code } = req.query;
-            if (code) {
-              console.log(code);
-              resolve(code);
+            const { receivedCode } = req.query;
+            if (receivedCode) {
+              console.log(receivedCode);
+              resolve(receivedCode);
```
Comment (tjallingt) [inline lib/AthomApi.js:88]:
```suggestion
            if (req.query.code) {
              console.log(req.query.code);
              resolve(req.query.code);
```

This is the name that appears in the url, we can't change that

File: lib/AthomApi.js [hunk 6]
```diff
@@ -98,11 +99,11 @@ class AthomApi {
           name: 'code',
```
Comment (tjallingt) [inline lib/AthomApi.js:99]:
```suggestion
          name: 'receivedCode',
```

needs to match the name you are destructuring from the result

File: lib/AthomApi.js [hunk 7]
```diff
@@ -169,9 +168,8 @@ class AthomApi {
     // find USB connected Homeys
     if (local) {
       const ifaces = os.networkInterfaces();
-      const candidates = [];
 
-      for (const ifaceId in ifaces) {
+      for (const ifaceId of Object.keys(ifaces)) {
         const adapters = ifaces[ifaceId];
         for (let i = 0; i < adapters.length; i++) {
           const adapter = adapters[i];
```
Comment (tjallingt) [inline lib/AthomApi.js:175]:
```suggestion
      for (const adapters of Object.values(ifaces)) {
        for (const adapter of Object.values(adapters)) {
```

File: lib/GitCommands.js [hunk 8]
```diff
@@ -99,7 +96,10 @@ class GitCommands {
 
     // Create an array based on line breaks, then filter any empty String out of it.
     return output.split(/\r\n|\r|\n/).filter(value => {
-      if (value) return value;
+      if (value) {
+        return value;
+      }
+      return false;
```
Comment (tjallingt) [inline lib/GitCommands.js:102]:
```suggestion
      return Boolean(value);
```

Github doesn't let me suggest it inline but you can do `filter(Boolean)` too... In most cases using something as a callback that is not intended to be used that way is a bad idea but in the case of `Boolean` I think its fine.

File: lib/App.js [hunk 9]
```diff
@@ -2394,7 +2397,6 @@ class App {
     appPath = path.join(appPath, appJson.id);
 
     try {
-      const stat = await statAsync(appPath);
       throw new Error(`Path ${appPath} already exists`);
     } catch (err) {
       if (err.code === undefined) throw err;
```
Comment (tjallingt) [inline lib/App.js:2402]: This change is not right. I believe this is checking whether a folder exists in the strangest way possible:
`statAsync` throws an error if the given path doesn't exist so, if it does the next line throws.
errors thrown by Node.JS build-in modules have a `.code` property so we re-throw if the error did not come from `statAsync`.

Just call `fs.existsSync` and throw the error if it doesn't exist...

File: lib/Util.js [hunk 10]
```diff
@@ -24,4 +31,14 @@ module.exports = class Util {
     throw new Error(`Unknown Algorithm: ${algorithm}`);
   }
 
+  static parseMacToDecArray(macAddress) {
+    const mac = [];
+    macAddress
+      .slice(0, 8)
+      .split(':') // TODO - is also a valid MAC address seperator
```
Comment (tjallingt) [inline lib/Util.js:38]: I know this is not new but a regex like `const MAC_LIKE = /([0-9A-F]{2}[:-]){5}([0-9A-F]{2})/i;` would be better I think...

File: lib/App.js (hunk 9 above)
Comment (mrlukasbos) [reply lib/App.js]: You are completely right in both ways: 1) the change is incorrect and 2) it worked in the strangest way possible.

**Comment by mrlukasbos** [comment]
I have tested the CLI, made some new apps, tried to run/validate them, logged out, logged in, selected new homeys, added driverse/flows/discovery strategies

**Review by tjallingt — APPROVED** [review]

**Pull request merged**